recipes/full_finetune_distributed.py

```diff
@@ -147,7 +147,11 @@ def setup(self, cfg: DictConfig) -> None:
         Sets up the recipe state correctly. This includes setting recipe attributes based
         on the ``resume_from_checkpoint`` flag.
         """
-        self._metric_logger = config.instantiate(cfg.metric_logger)
+        if self._is_rank_zero:
```

Comment:
you need to also update the cleanup method to check for rank zero before closing the logger